data/scripts/bin/update-hosts.sh.template

```diff
@@ -3,16 +3,24 @@ set -e
 
 BASE_URL="${SERVICE_BASE_URL}api/assisted-install/v2"
 
-while [[ "${cluster_status}" != "ready" ]]
+required_master_nodes=$(grep -m 1 required_master_nodes '/usr/local/bin/start-cluster-installation.sh' | awk -F"=" '{ print $2 }')
+required_worker_nodes=$(grep -m 1 required_worker_nodes '/usr/local/bin/start-cluster-installation.sh' | awk -F"=" '{ print $2 }')
+total_required_nodes=$(( ${required_master_nodes}+${required_worker_nodes} ))
+
+declare -a host_ids
+declare -i num_of_hosts
+while [[ "${num_of_hosts}" != "${total_required_nodes}" ]]
 do
-    echo "Waiting for cluster to become ready..." 1>&2
-    cluster_status=$(curl -s -S "${BASE_URL}/clusters" | jq -r .[].status)
-    echo "Cluster status: ${cluster_status}" 1>&2
+    echo "Waiting for ${total_required_nodes} required hosts..." 1>&2
+    host_ids=$(curl -s -S "${BASE_URL}/infra-envs/${INFRA_ENV_ID}/hosts" | jq -r .[].id)
```

Comment:
what if we got no response here? perhaps it needs to catch errors, empty responses, and retry. this can easily break.

Reply:
we ensure the response is not empty in line 16. same as done in the installer: https://github.com/openshift/installer/blob/f7731922a0f17a8339a3e837f72898ac77643611/data/data/agent/files/usr/local/bin/start-cluster-installation.sh.template#L33